packages/core/test/behaviors.test.ts

```diff
@@ -186,3 +298,189 @@ describe('observable behavior (fromObservable)', () => {
+          }
+        }
+      }
+    });
+
+    const machine = createMachine({
+      initial: 'waiting',
+      invoke: [
+        {
+          id: 'a',
+          src: fromPromise(() => Promise.resolve(42)),
+          onDone: {
+            // @ts-ignore TODO: fix types
+            actions: raise({ type: 'done' })
+          }
+        },
+        {
+          id: 'b',
+          src: childMachine
+        }
+      ],
+      states: {
+        waiting: {
+          on: {
+            done: 'success'
+          }
+        },
+        success: {}
+      }
+    });
+
+    const actor = interpret(machine).start();
+
+    await waitFor(actor, (s) => s.matches('success'));
+
+    const persistedState = actor.getPersistedState()!;
+
+    expect(persistedState.children.a.state).toEqual(
+      expect.objectContaining({
+        canceled: false,
+        data: 42
+      })
+    );
+
+    expect(persistedState.children.b.state).toEqual(
+      expect.objectContaining({
+        context: {
+          count: 55
+        },
+        value: 'start',
+        children: {
+          reducer: expect.objectContaining({
+            state: {
+              status: 'active'
+            }
+          })
+        }
+      })
+    );
+  });
+
+  it('should persist and restore a nested machine', () => {
+    const childMachine = createMachine({
+      initial: 'a',
+      states: {
+        a: {
+          on: {
+            NEXT: 'b'
+          }
+        },
+        b: {
+          on: {
+            LAST: 'c'
+          }
+        },
+        c: {}
+      }
+    });
+
+    const parentMachine = createMachine({
+      initial: 'idle',
+      states: {
+        idle: {
+          on: {
+            START: 'invoked'
+          }
+        },
+        invoked: {
+          invoke: {
+            id: 'child',
+            src: childMachine
+          },
+          on: {
+            NEXT: {
+              actions: sendTo('child', { type: 'NEXT' })
+            },
+            LAST: {
+              actions: sendTo('child', { type: 'LAST' })
+            }
+          }
+        }
+      }
+    });
+
+    const actor = interpret(parentMachine).start();
+
+    // parent is at 'idle'
+    // ...
+    actor.send({ type: 'START' });
+    // parent is at 'invoked'
+    // child is at 'a'
+    // ...
+    actor.send({ type: 'NEXT' });
+    // child is at 'b'
+
+    const persistedState = actor.getPersistedState()!;
+    const newActor = interpret(parentMachine, {
+      state: persistedState
+    }).start();
+    const newSnapshot = newActor.getSnapshot();
+
+    expect(newSnapshot.children.child.getSnapshot().value).toBe('b');
+
+    // Ensure that the child actor is started
+    // LAST is sent to parent which sends LAST to child
+    newActor.send({ type: 'LAST' });
+    // child is at 'c'
+
+    expect(newActor.getSnapshot().children.child.getSnapshot().value).toBe('c');
+  });
+
+  it('should return the initial persisted state of a non-started actor', () => {
```

Comment:
could we also add a test case that would check that the initial state of a child is available in the persisted state even before we start its parent?

Reply:
b75d480508b38f5b29b4749c9deec3478f8c1508